### PR TITLE
feat!: use common crm schema for list memberships. cleanup interface

### PR DESCRIPTION
**Pull request by tomkit**
product:
- Use CRM common schema for List Memberships API
- Make response interface match other CRM APIs

tech notes:
- Re-using remote provider --> database mapper (and unfortunately: legacy decisions)
- Need help DRYing up endpoint with TS generics mapping
- Reuse existing code to come up with the list of properties to fetch for Hubspot + Salesforce (so we get all properties for Hubspot and custom properties for Salesforce)

## Test Plan

Salesforce: https://gist.github.com/tomkit/3f3630b619a264c070379ded18aa8599
Hubspot: https://gist.github.com/tomkit/0a586c962f37f5c3156e1732eb051a07

- Salesforce contact/account/lead/opportunity list lists and list membership
- Hubspot contact list lists and list membership

Screenshot 2023-09-27 at 11 18 53 AM

## Deployment instructions

[Add any special deployment instructions here]

### CODE REVIEW

File: openapi/v2/crm/components/schemas/objects/list_membership.yaml [hunk 1]
```diff
@@ -1,76 +1,9 @@
...
-      IsEmailBounced: false
-      PhotoUrl: '/services/images/photo/003Dn00000TKj6wIAD'
-      Jigsaw: null
-      JigsawContactId: null
-      CleanStatus: 'Pending'
-      IndividualId: null
-required:
-  - id
-  - raw_data
+oneOf:
```
Comment (tomkit) [inline openapi/v2/crm/components/schemas/objects/list_membership.yaml:1]: Show schema of response for all common objects we support

File: openapi/v2/crm/paths/lists@{object_type}@{list_id}.yaml [hunk 2]
```diff
@@ -14,20 +14,21 @@ get:
...
-                    type: array
-                    items:
-                      $ref: ../components/schemas/objects/list_membership.yaml
-                  metadata:
-                    description: Metadata about the remote provider list view.
-                    $ref: ../components/schemas/objects/list_metadata.yaml
-                required:
-                  - results
-                  - metadata
+            type: object
```
Comment (tomkit) [inline openapi/v2/crm/paths/lists@{object_type}@{list_id}.yaml:17]: Make the pagination response signature consistent

File: packages/core/remotes/impl/hubspot/index.ts [hunk 3]
```diff
@@ -2039,7 +2041,7 @@ class HubSpotClient extends AbstractCrmRemoteClient implements MarketingAutomati
   public override async listLists(
     objectType: Exclude<CRMCommonObjectType, 'user'>,
     paginationParams: PaginationParams
-  ): Promise<PaginatedResult<ListMetadata>> {
+  ): Promise<PaginatedSupaglueRecords<ListMetadata>> {
```
Comment (tomkit) [inline packages/core/remotes/impl/hubspot/index.ts:2044]: Make the pagination signature consistent with others

File: packages/core/remotes/impl/hubspot/index.ts [hunk 4]
```diff
@@ -2060,45 +2062,55 @@ class HubSpotClient extends AbstractCrmRemoteClient implements MarketingAutomati
...
-      previous: null,
+      pagination: {
+        next: response.data['has-more']
+          ? encodeCursor({
+              id: response.data.offset,
+              reverse: false,
+            })
+          : null,
+        previous: null,
+        total_count: -1,
```
Comment (tomkit) [inline packages/core/remotes/impl/hubspot/index.ts:2080]: I don't believe there's a total count on this?

File: packages/core/remotes/impl/hubspot/index.ts [hunk 5]
```diff
@@ -2060,45 +2062,55 @@ class HubSpotClient extends AbstractCrmRemoteClient implements MarketingAutomati
...
       axios.get(`https://api.hubapi.com/contacts/v1/lists/${listId}/contacts/all`, {
         headers: {
           Authorization: `Bearer ${this.#config.accessToken}`,
         },
         params: {
+          property: propertiesToFetch,
           count: paginationParams.page_size,
           vidOffset: cursor?.id,
         },
+        paramsSerializer: {
```
Comment (tomkit) [inline packages/core/remotes/impl/hubspot/index.ts:2109]: Hubspot v1 (unlike v3) needs it in the form `&property=...&property=...`

File: packages/core/remotes/impl/salesforce/index.ts [hunk 6]
```diff
@@ -1398,30 +1398,33 @@ ${modifiedAfter ? `WHERE SystemModstamp > ${modifiedAfter.toISOString()} ORDER B
...
-    objectType: Exclude<CRMCommonObjectType, 'user'>,
+  public override async listListMembership<T extends ListCRMCommonObject>(
+    objectType: T,
     listId: string,
     paginationParams: PaginationParams
-  ): Promise<PaginatedResult<ListMember> & { metadata: ListMetadata }> {
-    const mappedObjectType = capitalizeString(objectType);
+  ): Promise<PaginatedSupaglueRecords<ListCRMCommonObjectTypeMap<T>> & { metadata: ListMetadata }> {
+    const commonObjectType = capitalizeString(objectType);
+    const propertiesToFetch = await this.getCommonPropertiesToFetch(objectType);
```
Comment (tomkit) [inline packages/core/remotes/impl/salesforce/index.ts:1425]: Reuse to fetch same properties as CRM common object list endpoints

File: packages/core/remotes/impl/salesforce/index.ts [hunk 7]
```diff
@@ -1438,33 +1441,56 @@ ${modifiedAfter ? `WHERE SystemModstamp > ${modifiedAfter.toISOString()} ORDER B
...
+        break;
+      case 'lead':
+        commonObjectRecords = response.records.map((record) => ({
+          ...fromSalesforceLeadToLead(record),
+        })) as ListCRMCommonObjectTypeMap<T>[]; // TODO: figure out types
+        break;
+      case 'opportunity':
+        commonObjectRecords = response.records.map((record) => ({
+          ...fromSalesforceOpportunityToOpportunity(record),
+        })) as ListCRMCommonObjectTypeMap<T>[]; // TODO: figure out types
```
Comment (tomkit) [inline packages/core/remotes/impl/salesforce/index.ts:1473]: Need help with types

**Review by lucasmarshall — APPROVED** [review]

**Pull request merged**